### PR TITLE
fix CI + update of stream package

### PR DESCRIPTION
**Pull request by jonasfj**
This should fix two intermittent bugs :) and get CI passing... + update a dependency (I suspect stream caused some deadlocks), so that's nice

### CODE REVIEW

**Comment by jonasfj** [comment]
The OSX CI job not running is because pmoore is offline :)

**Comment by jonasfj** [comment]
I can't reproduce the current error on macstadium... but we've certainly fixed a few bugs with this, so I suggest we land it asap.

File: worker/workertest/superseded_test.go [hunk 1]
```diff
@@ -4,19 +4,51 @@ import (
...
 	var c Case
+	var m sync.Mutex
 	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		m.Lock() // lock access to superseded
+		defer m.Unlock()
+
+		// Get taskId from which this is being called
 		taskID := r.URL.Query().Get("taskId")
+
+		// Create array of superseding taskIds, always start with it's own taskId
```
Comment (walac) [inline worker/workertest/superseded_test.go]: nit: "its own"

File: worker/workertest/superseded_test.go [hunk 2]
```diff
@@ -4,19 +4,51 @@ import (
...
 		supersedes := []string{taskID}
-		if c.Tasks[0].TaskID == taskID {
-			supersedes = append(supersedes, c.Tasks[1].TaskID)
+
+		// If we haven't made a superseded task yet we do that now
+		if superseded == -1 {
+			// Find index of this taskId
+			for index, task := range c.Tasks {
+				if task.TaskID == taskID {
+					superseded = index
```
Comment (walac) [inline worker/workertest/superseded_test.go]: shouldn't we add a `break` statement here?

File: worker/workertest/superseded_test.go [hunk 3]
```diff
@@ -28,6 +60,24 @@ func TestSuperseded(t *testing.T) {
...
+	verifyTask := func(i int) StatusAssertion {
+		return func(t *testing.T, q *queue.Queue, status queue.TaskStatusStructure) {
+			// Sanity check for status structure
+			assert.Equal(t, c.Tasks[i].TaskID, status.TaskID, "Expected taskId to match")
+			if i == superseded {
+				// If this is the task that was superseded, we assume it was resolved as such
+				runID := len(status.Runs) - 1
+				assert.Equal(t, status.Runs[runID].ReasonResolved, runtime.ReasonSuperseded.String(), "expected superseded")
+			} else {
+				// If this task was not superseded is should be successful
```
Comment (walac) [inline worker/workertest/superseded_test.go]: nit: "it should"

File: worker/workertest/superseded_test.go (hunk 2 above)
Comment (jonasfj) [reply worker/workertest/superseded_test.go]: No harm either way.

**Pull request merged**